### PR TITLE
Add Output Options for Bilinear Function Format

### LINKED ISSUE #13988: Postprocessor Python Script Outputting PiecewiseBilinear Format
## Reason

In `python/postprocessing` is a script that combines "csv" files together. This script works well on vector postprocessor output to put it together. However, I'd like to extend that to also output the csv files in a format that a PiecewiseBilinear function can read in.  

An example might help describe it:  
I run a simulation and vector postprocess the wall temperature as a function of time and position. I then take that wall temperature and impose it as a boundary condition on another simulation using a PiecewiseBilinear function for time and position.

## Design
A few changes:
  1. Script imports Pandas to easily transpose column and row.
  1. Option to read "*time.csv" vector out file linking timestep to time.
  1. Option to actual proceed with formating csv as a Bilinear csv file.

The previous functionality of the script is preserved and this only added new functionality to it, all with optional command line flags. I also believe that Pandas is already in MOOSE's environment, but I honestly didn't check...  

Command to run script to use new functionality:  
```
python ../combine_csv.py -b -t -o out_bilin.csv -w -x y -y wall_temp -s 1 vec_wall_temp_
```
The new options are "-b" and "-t".

## Impact
I believe none. However, the script will need to be manually ran to check it. Also, I plan on making the changes, so stay tuned for a PR. 😉

### PR DESCRIPTION
**Pull request by elementx54**
When combining csv files, the output may now match the input for the
`PiecewiseBilinear` function.

Closes #13988

Design Change:  
The script require Pandas now.  All other additions are optional and are behind if statements.

Testing:  
I did not see a test for the `python/postprocessing` directory. Please let me know if I missed it and I'll make one.

### CODE REVIEW

**Review by permcody — DISMISSED** [review]
This looks fine to me and yes "pandas" should be available in our environment. Tagging @milljm to get a thumbs up if this is indeed a prerequisite when not using the pre-built packages.

It would be awesome if you could add a unit test. We are getting better at adding tests for the Python utilities we create too. We use "unittest" and there are several examples in the python directory. Just copy'n'paste from another example and give it a shot. You are correct, there isn't currently a test for the existing script.

**Comment by elementx54** [comment]
I'm looking for an example to copy and modify to make a new test for `combine_csv.py` but I can't find any. The other files seem to be set up to have a class that could then be imported and tested directly. `combine_csv.py` seems more like a script than a full fledged class. Perhaps this would be a good time to switch it to the "scripts" directory instead of "python/postprocessing" directory where it is the sole file.  

Do scripts also require tests? Is there a test that just runs a command line argument?

**Comment by milljm** [comment]
@permcody @elementx54 Pandas is available in our moose-environment package only, and it (the moose-environment package) would certainly be a prerequisite for this to work.

**Comment by elementx54** [comment]
Or, the other option, is I could make it into a class and bring it in line with the other python tools in "python". Then I could just copy the tests which already exist for testing python classes. Which one would you like?

**Comment by permcody** [comment]
@aeslaughter - any advice for Adam?

**Comment by aeslaughter** [comment]
Take a look at python/mooseutilts/tests for an example of how we are doing python testing, sere [test_AutoPropertyMixin.py](https://github.com/idaholab/moose/blob/next/python/mooseutils/tests/test_AutoPropertyMixin.py) for a good specific example.

We use the python [unittest](https://docs.python.org/3.7/library/unittest.html) and you will need to also create a tests file/spec, as in [tests](https://github.com/idaholab/moose/blob/next/python/mooseutils/tests/tests).

Finally, if you are good with python can you make sure that this new code is Python3 capable, we will be switching over very soon.

**Comment by elementx54** [comment]
Alright sounds good.  

I'll make it into its own class so it can be imported for testing. I'm going to preserve the command line execution though since I like that. I'll have to get to it after I get two documents out, so don't expect a quick turn around on this.  

Be in touch. 👋

**Comment by elementx54** [comment]
@aeslaughter I got everything ready and it is working with Python3. However, I can't get it to run with `run_tests` as I get "python 2.7 is required to run the test harness".  

Should I push up what I have so you can look at it? Or is there a Python3 branch I could test it on?

**Comment by milljm** [comment]
@elementx54 If you go to [civet.inl.gov](civet.inl.gov), and click on your PR, or click here: [PR](https://civet.inl.gov/pr/9231/). You should be presented with a host of additional tests you can instruct Civet to perform.

Among that list, is an 'Extensive Python3 Tests' target. Feel free to check that, and hit the submit button at the bottom of the page. Do know, that _that_ target will probably fail. But it will run with python3.

Failures are expected (in this case). So again, feel free to enable that Extensive Python3 target...

I would do it for you, but it's good to learn!

**Comment by elementx54** [comment]
Well, I enabled Extensive Python3 testing and it failed:
```
postprocessing/tests.combine_csv:   File "test_combine_csv.py", line 13, in <module>
postprocessing/tests.combine_csv:     import pandas
postprocessing/tests.combine_csv: ImportError: No module named pandas
```

I wasn't expecting that. Is pandas not available in MOOSE for Python3?  

Also, the full commit (ready for merge) is available if you want to check it out even though the Pyhon3 is going to make it fail tests. I also noticed there is a python-3 branch, should I do some local tests using that?

**Comment by milljm** [comment]
Its okay... I knew it would fail. I was only interested in 'how much'. The other failures about libMesh/PETSc are also not your fault.

The failures are occurring due to the shebang still calling for Python2, when all there is available now, is Python3 (pandas _is_ available, but only in python3).

File: python/postprocessing/combine_csv.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
```
Comment (milljm) [inline python/postprocessing/combine_csv.py]: I would change this to simply `python`.

File: python/postprocessing/tests/test_combine_csv.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
```
Comment (milljm) [inline python/postprocessing/tests/test_combine_csv.py]: Change to `python` as my previous comment.

**Comment by aeslaughter** [comment]
Do you need this in the repository soon, we are working converting to python3 and hope to switch over very soon (1 or 2 weeks). Could this wait until then?

**Comment by elementx54** [comment]
Yes, this can wait. I already used it to convert my stuff. I just thought since it was useful to me, that others might find it as well. So, I'll see you guys in a few weeks. 👋

**Comment by elementx54** [comment]
> 
> 
> This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

Oh stalebot, you so funny. They told me to wait till Python 3 was ready and I've been waiting. I don't think the conversion to Python 3 is in yet.

**Comment by milljm** [comment]
No stale bot!

**Comment by elementx54** [comment]
Alright, I've got the unit test in and it passes on my local machine. Hopefully all goes well here, too.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**